client.go

```diff
@@ -50,26 +80,35 @@ type userResponse struct {
 	UserId   string `json:"Id"`
 }
 
+// Login authenticates a user into the server provided in Client.
+// If the login is successful, the access token is stored for future API calls.
 func (c *Client) Login(username, password string) error {
-	body := map[string]string{}
-	body["Username"] = username
-	body["PW"] = password
+	body := map[string]string{
+		"Username": username,
+		"PW":       password,
+	}
 
-	b := &bytes.Buffer{}
-	_ = json.NewEncoder(b).Encode(body)
+	bodyBytes, err := json.Marshal(body)
+	if err != nil {
+		return err
+	}
 
-	auth := c.authHeader()
-	req, err := http.NewRequest("POST", c.BaseURL+"/Users/authenticatebyname", b)
+	u, err := url.JoinPath(c.BaseURL().String(), "/Users/authenticatebyname")
 	if err != nil {
-		return fmt.Errorf("failed to login: %v", err)
+		return fmt.Errorf("unable to parse url path: %w", err)
 	}
 
-	req.Header.Set("X-Emby-Authorization", auth)
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodPost, u, io.NopCloser(bytes.NewBuffer(bodyBytes)))
```

Comment:
Oh- one more thing- Shouldn't `io.NopCloser(...)` just become `bytes.NewReader(bodyBytes)`?

Reply:
Looking into this more I think I liked your original Encode code more than marshaling and then converting. It seems to do the same thing in less code. Reverted this piece but kept the error check. 

The requests that use `post()` method still use the json marshal-er but that can be changed in the future. I'm not aware of a functional difference (maybe one way is more performant than the other), but for code consistency they should probably both either use marshal/unmarshal OR encode/decode.